### DISCUSSION TITLE
[Feature request] Switch player audio track with URL or JS API

### DISCUSSION THREAD

**Discussion by PixelsCommander** [vimeo/player.js#969] [Ideas]
Any chance to switch the player audio track with URL or JS API? The use case is a multilingual website with tracks being translations. When we switch to another language we want track to be set based on the language.

For example, [here](https://portal.gitnation.org/contents/7-patrones-de-typescript-que-deberias-estar-utilizando) we have a conference talk video with a Spanish audio track and we want to switch to Spanish automatically when visiting the Spanish translation of the page.

Switching subtitles in the same way would be also a part of the problem

**Comment by bdougherty** [comment]
Sorry, we totally dropped the ball on this. We have a ticket to add this capability, but it hasn't been scheduled yet. I'll try to get that figured out so we can make this available soon.

**Reply by PixelsCommander** [reply]
@bdougherty thank you so much! Any estimations when this could be available?

**Reply by rkrishnan8594** [reply]
We plan to release a new `audio_track` embed parameter soon - hopefully within the next 1-2 weeks.

**Reply by PixelsCommander** [reply]
@rkrishnan8594 What is the best way to get updated about the parameter rollout as soon as it happens? Sorry for the disturbance

**Reply by rkrishnan8594** [reply]
Hi @PixelsCommander - no worries at all. We've actually decided to change our solution to this problem; rather than adding a new embed parameter, we're working on auto selecting an audio track based on the user's browser language (i.e. the value set in the `Accept-Language` header).

**Reply by PixelsCommander** [reply]
I have to express our team's concern about the new approach. The headers are not straightforward and in case a developer has UI switch EN / ES  as we do on https://gitnation.com/contents/you-cant-use-hooks-conditionally-or-can-you then the user might end up having inconsistent language video / page. The parameter was perfect for us while the headers cover our use case partially. Anyway, thanks for making any progress on this!

**Reply by rkrishnan8594** [reply]
Thank you for the feedback, @PixelsCommander. We feel that respecting the viewer's browser language would have the widest impact, as it would automatically choose the viewer's preferred language without any changes needed on the embed. We'd still like to eventually add programmatic support for switching audio tracks.

**Reply by robhrt7** [reply]
Just chipping in with some more context why we feel having a SDK control over the language switch is more important than browser header based solution that decides default behaviour for users.

Videos always have original/native audio track and sometimes alternative audio tracks and subtitles. In current age, the alternative tracks are likely auto-generated (with some form of AI), not professionally dubbed, on average lower quality that original. Given that alternative translation is not always the best and may not be original track, same as with movies, people quite often may prefer original track higher quality track + subtitles in their language.

Default auto-switching of audio tracks from original to alternative based on system/browser language preferences, may lead to some regressions, when video authors expect that viewers experience content in original form. I would say that in our case we'd also rather default to original recording if no SDK control give, retaining real voices of presenters in the recording by default, and show UI box in our app suggesting viewers can adjust player settings.

There may be many different use cases, and programmatic ability (with SDK) to pick audio track and subtitles would allow products to provide more tailored setup for their content.

**Comment by rkrishnan8594** [comment]
Just to follow up here - we've added the `player.selectAudioTrack()` method to the SDK.
